app/src/main/java/com/mapbox/maps/testapp/utils/SimulateRouteLocationProvider.kt

```diff
@@ -0,0 +1,57 @@
+package com.mapbox.maps.testapp.utils
+
+import android.os.Handler
+import android.os.Looper
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.maps.plugin.locationcomponent.LocationConsumer
+import com.mapbox.maps.plugin.locationcomponent.LocationProvider
+import com.mapbox.turf.TurfMeasurement
+import java.util.concurrent.*
+
+/**
+ * A location provider implementation that takes in a line string as route and animate the location
+ * updates along the route.
+ */
+class SimulateRouteLocationProvider(
+  val route: LineString,
+) : LocationProvider {
+  private val locationConsumers = CopyOnWriteArraySet<LocationConsumer>()
+  private val handler = Handler(Looper.getMainLooper())
+  override fun registerLocationConsumer(locationConsumer: LocationConsumer) {
```

Comment:
If I register location consumer twice, I will get locations twice faster?

Reply:
yeah, it's set. But after adding `emitFakeLocations` called inside this function. And it added new postDelay to our handler.

Reply:
yes, added a flag to avoid triggering emitFakeLocations multiple times, thanks for catching it.